cmd/snaplock/runinhibit/inhibit.go

```diff
@@ -283,3 +278,113 @@ func RemoveLockFile(snapName string) error {
 
 	return nil
 }
+
+// needed for better mocking
+type ticker interface {
+	Wait()
+	Stop()
+}
+
+type timeTicker struct {
+	*time.Ticker
+}
+
+func (t *timeTicker) Wait() {
+	if t.Ticker != nil {
+		<-t.Ticker.C
+	}
+}
+
+var newTicker = func(interval time.Duration) ticker {
+	return &timeTicker{time.NewTicker(interval)}
+}
+
+// WaitWhileInhibited blocks until snap is not inhibited anymore returning
+// a locked hint file lock.
+//
+// "notInhibted" and "inhibited" callbacks are called with the hint file lock held.
+// If inhibited callback returns true, WaitWhileInhibited returns instantly without
+// waiting for the snap not being inhibited.
```

Comment:
Something like:
```go
// The notInhibited callback is called with the hint file lock is acquired if the snap is not inhibited.
// The inhibited callback is called with the hint file lock held every time the lock is acquired and the snap is still inhibited.
// If inhibited returns cont==false, it waits for the given interval and retries to acquire the lock.
// Returns with the hint file lock held.
```

But that may be too much detail, not sure. I think it's useful to point out that either the `inhibited` callback is called or the `notInhibited` callback, but not both, unless the `inhibited` callback signals to retry.